c++/src/kj/async.c++

```diff
@@ -332,39 +370,213 @@ private:
   struct Impl;
   Impl* impl;
 #endif
+
+  void run();
+
+  bool isReset() { return main == nullptr; }
 };
 
-}
+}  // namespace _
+
+#if __linux__
+// TODO(someday): Support core-local freelists on OSs other than Linux. The only tricky part is
+//   finding what to use instead of sched_getcpu() to get the current CPU ID.
```

Comment:
On Windows you'd use `GetCurrentProcessorNumberEx` (idk about OSX) but we could probably leave that for another PR later.

Reply:
Oh nice, that even identifies physical cores vs. hyperthreads. For our use case we'd care about physical cores...

Reply:
I don't think so? Windows [processor groups](https://docs.microsoft.com/en-us/windows/win32/procthread/processor-groups) are static groups of 64 logical processors used by Windows for scheduling, not physical cores.

Reply:
Oh, darn... I saw them use the term "logical processor" which is commonly used as a generic term for "hyperthread", and assumed that distinguishing hyperthreads from cores was the whole point, but it appears not. Darn.